build: find pre-built heimdal build tools in case of embedded heimdal

Rebased for 4.23.13 from the buildroot 4.22 patch (Uri Simchoni / Bernd Kuhls):
host compile_et and asn1_compile are used when excluded from bundled libs.

Upstream: https://bugzilla.samba.org/show_bug.cgi?id=14164

--- a/wscript_configure_embedded_heimdal	2026-10-16 18:00:22.412428170 +0200
+++ b/wscript_configure_embedded_heimdal	2026-10-16 18:00:22.632282063 +0200
@@ -16,3 +16,14 @@
 
 conf.define('HAVE_KRB5_INIT_CREDS_STEP', 1)
 conf.define('HAVE_GSS_KEY_VALUE_SET_DESC', 1)
+
+def check_system_heimdal_binary(name):
+    if conf.LIB_MAY_BE_BUNDLED(name):
+        return False
+    if not conf.find_program(name, var=name.upper()):
+        return False
+    conf.define('USING_SYSTEM_%s' % name.upper(), 1)
+    return True
+
+check_system_heimdal_binary("compile_et")
+check_system_heimdal_binary("asn1_compile")
